lightning/src/ln/channelmonitor.rs

```diff
@@ -465,10 +466,24 @@ pub(crate) enum InputMaterial {
 		locktime: u32,
 	},
 	LocalHTLC {
-		witness_script: Script,
-		sigs: (Signature, Signature),
+		their_sig: Signature,
 		preimage: Option<PaymentPreimage>,
 		amount: u64,
+		feerate_per_kw: u64,
+		their_to_self_delay: u16,
+		htlc: HTLCOutputInCommitment,
+		delayed_payment_key: PublicKey,
+		revocation_key: PublicKey,
+		a_htlc_key: PublicKey,
+		b_htlc_key: PublicKey,
+		per_commitment_point: PublicKey,
+		htlc_base_key: SecretKey,
+	},
+	Funding {
+		local_tx_remote_signed: LocalCommitmentTransaction,
+		funding_key: SecretKey,
```

Comment:
Don't pass the SecretKeys - you can get them out of the ChannelKeys already with ChannelKeys::funding_key() or ChannelKeys::htlc_base_key().

Reply:
Hmmmm what do you suggest exactly ? To introduce yet ChannelKeys in OnchainTxHandler ?

Reply:
Yea, just pass in a reference to the ChannelMonitor ChannelKeys in OnchainTxHandler::block_connected() and pass it down. We can rework it later, but its so awkward to copy private keys around like this.